### PR TITLE
[chore] SDP - adding the profiler labels

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Adding a couple of labels on the shardedDDP side

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by min-xu-ai — APPROVED** [review]
awesome!

File: fairscale/nn/data_parallel/sharded_ddp.py [hunk 1]
```diff
@@ -199,25 +200,26 @@ def forward(self, *inputs: Any, **kwargs: Any) -> Any:
         backward pass for gradient reduction to the proper ranks.
         """
 
-        # Deferred initialization, or change detection
-        needs_setup = len(self._grad_hooks) == 0 and self.training
+        with profiler.record_function("sdp_forward"):
```
Comment (anj-s) [inline fairscale/nn/data_parallel/sharded_ddp.py:203]: Same comment as the other PR. Lets add some namespace information?
